enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/mtls/MtlsUtils.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.enforcer.security.mtls;
+
+import com.google.protobuf.ByteString;
+import org.apache.commons.codec.binary.Base64;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Map;
+
+/**
+ * Common functions for certificates.
+ */
+public class MtlsUtils {
+    public static boolean isCertificate(String cert) {
+        return cert.contains("BEGIN") && cert.contains("CERTIFICATE") && cert.contains("END");
+    }
+
+    public static String getCertContent(String cert) {
+        return getCertContent(cert, false);
+    }
+    public static String getCertContent(String cert, boolean isEncoded) {
+        if (isEncoded) {
+            try {
+                cert = URLDecoder.decode(cert, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                throw new SecurityException(e);
+            }
+        }
+        String content = cert.replaceAll("-----BEGIN CERTIFICATE-----", "")
+                .replaceAll("-----END CERTIFICATE-----", "");
+        return content.trim();
+    }
+
+    public static KeyStore createTrustStore(Map<String, ByteString> clientCertMap) throws KeyStoreException {
+        KeyStore trustStore = KeyStore.getInstance("JKS");
+        try {
+            trustStore.load(null, null);
+        } catch (IOException | NoSuchAlgorithmException | CertificateException e) {
+            throw new SecurityException(e);
+        }
+        X509Certificate x509cert;
+        for (Map.Entry<String, ByteString> entry : clientCertMap.entrySet()) {
+            String alias = entry.getKey();
+            String cert = entry.getValue().toStringUtf8();
+            String certContent = getCertContent(cert);
+            try {
+                x509cert = getX509Cert(certContent);
+            } catch (java.security.cert.CertificateException e) {
+                throw new SecurityException(e);
+            }
+            trustStore.setCertificateEntry(alias, x509cert);
+        }
+        return trustStore;
+    }
+
+    public static X509Certificate getX509Cert(String certContent)
+            throws CertificateException {
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        byte[] bytes = Base64.decodeBase64(certContent);
```

Comment:
What if the string cannot be base64 decoded?
And is there any concerns regarding padding?

Reply:
If the string cannot be base64 decoded, it will throw an exception and print a debug log.

Reply:
Regarding padding concerns, I followed the approach in carbon-apimgt. [getClientCertificateFromHeader()](https://github.com/wso2/carbon-apimgt/blob/0b0a5884d92188c196a4c07d20bdec810a26b757/components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/Utils.java#L459)